ecosystem/indexer-grpc/indexer-grpc-utils/src/counters.rs

```diff
@@ -113,3 +131,227 @@ pub static TRANSACTION_UNIX_TIMESTAMP: Lazy<GaugeVec> = Lazy::new(|| {
     )
     .unwrap()
 });
+
+#[cfg(not(feature = "indexer_grpc_logging"))]
+pub fn log_grpc_step(
+    service_type: &str,
+    step: IndexerGrpcStep,
+    _start_version: Option<i64>,
+    end_version: Option<i64>,
+    _start_version_timestamp: Option<&Timestamp>,
+    end_version_timestamp: Option<&Timestamp>,
+    // Duration from the start of the batch to completing the IndexerGrpcStep.
+    // I chose to log this instead of individual step durations so that the whole processing duration is captured.
+    duration_in_secs: Option<f64>,
+    size_in_bytes: Option<usize>,
+    num_transactions: Option<i64>,
+    _request_metadata: Option<IndexerGrpcRequestMetadata>,
+) {
+    if let Some(duration_in_secs) = duration_in_secs {
+        DURATION_IN_SECS
+            .with_label_values(&[service_type, step.get_step(), step.get_label()])
+            .set(duration_in_secs);
+    }
+    if let Some(num_transactions) = num_transactions {
+        NUM_TRANSACTIONS_COUNT
+            .with_label_values(&[service_type, step.get_step(), step.get_label()])
+            .set(num_transactions);
+    }
+    if let Some(end_version) = end_version {
+        LATEST_PROCESSED_VERSION
+            .with_label_values(&[service_type, step.get_step(), step.get_label()])
+            .set(end_version);
+    }
+    if let Some(end_version_timestamp) = end_version_timestamp {
+        let end_txn_timestamp_unixtime = timestamp_to_unixtime(end_version_timestamp);
+        TRANSACTION_UNIX_TIMESTAMP
+            .with_label_values(&[service_type, step.get_step(), step.get_label()])
+            .set(end_txn_timestamp_unixtime);
+    }
+    if let Some(size_in_bytes) = size_in_bytes {
+        TOTAL_SIZE_IN_BYTES
+            .with_label_values(&[service_type, step.get_step(), step.get_label()])
+            .set(size_in_bytes as i64);
+    }
+}
+
+#[cfg(feature = "indexer_grpc_logging")]
```

Comment:
Feature means that you have to create a separate binary for this. I'd not use feature flag.